Paintroid-450 : Add share menu option in paint view

vishad-tyagi opened the pull request:
*Please enter a short description of your pull request and add a reference to the Jira ticket.*

Description
Add "share" option to the paint view overflow menu.
This should open the phone native share window.


## New Features and Enhancements
- New Features and Enhancements
## Refactorings and Bug Fixes
- Refactorings and Bug Fixes

## Checklist

#### Your checklist for this pull request
Please review the [contributing guidelines](https://github.com/Catrobat/Paintroid/blob/develop/README.md) and [wiki pages](https://github.com/Catrobat/Catroid/wiki/) of this repository.

- [ ] Include the name of the Jira ticket in the PR’s title
- [ ] Include a summary of the changes plus the relevant context
- [ ] Choose the proper base branch (*develop*)
- [ ] Confirm that the changes follow the project’s coding guidelines
- [ ] Verify that the changes generate no compiler or linter warnings
- [ ] Perform a self-review of the changes
- [ ] Verify to commit no other files than the intentionally changed ones
- [ ] Include reasonable and readable tests verifying the added or changed behavior
- [ ] Confirm that new and existing unit tests pass locally
- [ ] Check that the commits’ message style matches the [project’s guideline](https://github.com/Catrobat/Catroid/wiki/Commit-Message-Guidelines)
- [ ] Stick to the project’s gitflow workflow
- [ ] Verify that your changes do not have any conflicts with the base branch
- [ ] After the PR, verify that all CI checks have passed
- [ ] Post a message in the *#paintroid* [Slack channel](https://catrobat.slack.com) and ask for a code reviewer

Lenkomotive reviewed (changes requested):
The share package seems to be deprecated see [here](https://pub.dev/documentation/share/latest/index.html). Please always check if packages are maintained before adding them. 

Also- add tests please :)

vishad-tyagi commented:
@Lenkomotive I have changed the package to share_plus. and have added the tests also. while Native share windows cannot be tested in Flutter widget tests because these tests run in a Dart VM, that does not interact with native components. I have implemented a workaround of checking the entire widget tree before and after, looking for any unexpected behaviour and error dialog. If anything else needs to be changed let me know.
I will be careful in checking the package version from next time 😅

msesko commented on packages/features/workspace_screen/test/widget/workspace_screen_test.dart:
```diff
@@ -47,6 +47,25 @@ void main() {
+    await tester.pumpWidget(sut);
+
+    await tester.tap(find.byIcon(Icons.more_vert));
+    await tester.pumpAndSettle(); // Wait for the menu to open
```
remove comment please

vishad-tyagi replied:
I have removed the comment

msesko commented on packages/features/workspace_screen/lib/src/components/top_bar/overflow_menu.dart:
```diff
@@ -132,6 +139,10 @@ class _OverflowMenuState extends ConsumerState<OverflowMenu> {
   }
 
+  void _shareContent() {
+    Share.share('Check out this great app!');
```
Do we leave 'Check out this great app!' or should we write something more specific?

vishad-tyagi replied:
Something like the Catrobat website link? Or if you have something else in mind please let me know, and I will add it
Thank you

msesko reviewed (changes requested):
pls check my comments. thanks

juliajulie95 commented:
Thank you for the PR!
Please make sure the actual image that was drawn is shared with the share button, not just a text! :)

vishad-tyagi commented:
@juliajulie95 I intended to do the same initially only but was waiting for additional info like the default format of shared images, whether manual image compression is required or not etc 🙂 , I tried to get more info on the Jira ticket only. So I ended up placing a text-only
Just let me know this info, I will get to work in some time (My uni end sem exams are going on 😅)

juliajulie95 commented:
Sorry for that.
Please share the currently displayed image. You might have to save it as a temp and then share that file

bhav-khurana commented:
Are we still working on this ticket? @juliajulie95 @msesko 
I can complete it if anything is pending
